### PR TITLE
05/integration

### PR DESCRIPTION
**Pull request by Leandro97**

### CODE REVIEW

File: stant-album/ViewController.swift [hunk 1]
```diff
@@ -15,8 +15,8 @@ class ViewController: UIViewController {
     @IBOutlet weak var recordLabel:         UILabel!
     @IBOutlet weak var newAlbumButton:      UIButton!
     
-    var albumGenerator            = AlbumGenerator()
-    let child = SpinnerViewController()
+    var image                     = UIImage()
+    let loadSpinner                     = SpinnerViewController()
```
Comment (liuvasconcelos) [inline stant-album/ViewController.swift]: Identation

File: stant-album/ViewController.swift [hunk 2]
```diff
@@ -44,46 +44,63 @@ class ViewController: UIViewController {
     
     override func viewDidAppear(_ animated: Bool) {
         gradient.frame = self.view.safeAreaLayoutGuide.layoutFrame
-        child.view.frame = self.view.safeAreaLayoutGuide.layoutFrame
+        loadSpinner.view.frame = self.view.safeAreaLayoutGuide.layoutFrame
```
Comment (liuvasconcelos) [inline stant-album/ViewController.swift:48]: Identation

File: stant-album/ViewController.swift [hunk 3]
```diff
@@ -44,46 +44,63 @@ class ViewController: UIViewController {
...
     }
 
     @IBAction func getNewAlbumAction(_ sender: Any) {
         self.getNewAlbum()
     }
     
+    func makeAlert(errorTitle: String, errorMessage: String) {
```
Comment (liuvasconcelos) [inline stant-album/ViewController.swift]: make or show? auhaahuaha

File: stant-album/ViewController.swift [hunk 4]
```diff
@@ -44,46 +44,63 @@ class ViewController: UIViewController {
...
+        let alert = UIAlertController(title: errorTitle, message: errorMessage, preferredStyle: UIAlertController.Style.alert)
+        alert.addAction(UIAlertAction(title: "OK", style: UIAlertAction.Style.default, handler: nil))
+        self.present(alert, animated: true, completion: nil)
+    }
+    
     func getNewAlbum() {
-        let albumIndex      = Int.random(in: 0..<albumGenerator.albumList.count)
-        let newAlbum: Album = albumGenerator.albumList[albumIndex]
-        self.updateFields(newAlbum)
+        makeRequest(getFromURL: "https://fakerapiexample.herokuapp.com/album", requestCompletionHandler: { data, error in
```
Comment (liuvasconcelos) [inline stant-album/ViewController.swift]: I know that you only have one endpoint to consume, but always try to separate variable and make code more legible. In this case, you can put this url on a separated constant. 

let apiUrl: "https.: /////"

Comment (liuvasconcelos) [inline stant-album/ViewController.swift]: Do you remember what we talked about separate responsibilities? 
Try to put it in another class (like a presenter, or just an api data source) and make view controller haddler only view stuffs (what happens when it is success or failure).

**Review by liuvasconcelos — CHANGES_REQUESTED** [review]

File: stant-album/ViewController.swift (hunk 1 above)
Comment (Leandro97) [reply stant-album/ViewController.swift]: Done.

File: stant-album/ViewController.swift (hunk 4 above)
Comment (Leandro97) [reply stant-album/ViewController.swift]: Done.

File: stant-album/ViewController.swift (hunk 2 above)
Comment (Leandro97) [reply stant-album/ViewController.swift:48]: Done.

File: stant-album/ViewController.swift (hunk 3 above)
Comment (Leandro97) [reply stant-album/ViewController.swift]: Function name changed.

File: stant-album/ViewController.swift (hunk 4 above)
Comment (Leandro97) [reply stant-album/ViewController.swift]: Done

**Pull request merged**